streamer/src/nonblocking/quic.rs

```diff
@@ -1167,6 +1167,8 @@ async fn handle_connection(
                         CONNECTION_CLOSE_CODE_INVALID_STREAM.into(),
                         CONNECTION_CLOSE_REASON_INVALID_STREAM,
                     );
+                    stats.total_streams.fetch_sub(1, Ordering::Relaxed);
+                    stream_load_ema.update_ema_if_needed();
```

Comment:
PR says update metrics but this is a logic change?

Reply:
This is to restore the logic which is was broken in commit: 2be7c2e8c85b8363270d6c1eb247a0d0e78c6e84. We need to do the correct bookkeeping of the total_streams and ema before breaking the connection loop. For example, 
we did the following

        stats.total_streams.fetch_add(1, Ordering::Relaxed);
before the inner loop.

After the inner loop, we need to correct these counters.
The direct break to the outer loop cause these to be missed.

Reply:
Yep, I understand. What I'm saying is that the PR title says "update metrics" but the most important thing in this PR is arguably updating the EMA.

Having both changes in the same PR is fine of course, but please update PR/commit to reflect what's actually being changed. Also we're backporting this to 2.1 yeah?

Reply:
okay -- updated the title to include ema.